train.py

```diff
@@ -176,33 +158,39 @@ def model_fn(features, labels,
             config=config,
             params=hparams)
 
-    if args.valid:
-        train_spec = tf.estimator.TrainSpec(
-            input_fn=lambda params: input_fn(
-                args.train, args.vocab, args.norm, num_channels=args.num_channels,
+    def create_input_fn(mode):
+        if args.t2t_format:
+            return lambda params: utils.input_fn_t2t(args.train, mode, hparams,
                 batch_size=params.batch_size if 'batch_size' in params else args.batch_size,
-                num_epochs=args.num_epochs, binf2phone=None, num_parallel_calls=args.num_parallel_calls,
-                max_frames=args.max_frames, max_symbols=args.max_symbols),
+                num_epochs=args.num_epochs if mode == tf.estimator.ModeKeys.TRAIN else 1,
+                num_parallel_calls=64 if args.tpu_name and args.tpu_name != 'fake' else args.num_parallel_calls,
+                max_frames=args.max_frames, max_symbols=args.max_symbols)
+        else:
+            return lambda params: utils.input_fn(
+                args.valid if mode == tf.estimator.ModeKeys.EVAL and args.valid else args.train,
+                args.vocab, args.norm, num_channels=args.num_channels,
+                batch_size=params.batch_size if 'batch_size' in params else args.batch_size,
+                num_epochs=args.num_epochs if mode == tf.estimator.ModeKeys.TRAIN else 1,
+                num_parallel_calls=args.num_parallel_calls,
```

Comment:
Shouldn't t2t and non-t2t versions handle `num_parallel_calls` in the same way?

Reply:
Why not? The default value should naturally correspond to the number of cores.

Reply:
I mean that t2t version sometimes fallsback to 64, while non-t2t version always uses values from args.

Reply:
Inside `tensor2tensor` it is set to 64 if TPU is used.